### PR TITLE
Weekly habits

### LINKED ISSUE #6: BE dashboard page query: show last 7 days
- [x] moods

- [x] habits completed

### PR DESCRIPTION
**Pull request by WMudgeEllis**
# Change Log

- Closes #6 

## What I did:

- Added weekly habits to fetchUser query (fetchUser -> habits -> weeklyHabitEntries). Will return an array of all habits (even if there are no entries) then an array of habit entries
- removed date field from habit entries, will need to migrate after pulling.

## Things to consider:

- Habit query will need to be tweaked if/when we move to allowing users to create habits (so that a user cant see other user's habits).

## Test coverage

- [x] 100% Model Testing
- [] 100% Request Testing

If below 100%, what still needs testing:

- seems to largely be sad path/error testing in graphql files.

### CODE REVIEW

File: spec/requests/graphql/queries/fetch_user_weekly_habits_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      fetchUser{
+        id
+    		habits {
+    		  id
+          description
+          weeklyHabitEntries {
+            status
+            date
+          }
+    		}
```
Comment (WMudgeEllis) [inline spec/requests/graphql/queries/fetch_user_weekly_habits_spec.rb]: no idea why this one is floating off in space, it's in the right place on local.

File: app/graphql/types/habit_entry_type.rb [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
     field :user_id, Integer, null: true
     field :habit_id, Integer, null: true
     field :created_at, GraphQL::Types::ISO8601DateTime, null: false
     field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
+    field :date, String, null: false
+
+    def date
+      object.created_at
+    end
+
```
Comment (TannerDale) [inline app/graphql/types/habit_entry_type.rb:14]: We can remove updated at

File: db/seeds.rb [hunk 3]
```diff
@@ -126,7 +117,6 @@
 
 (1..10).to_a.each do |num|
   h10.habit_entries.create!(
-    date: Date.today - num,
```
Comment (Lmcourt) [inline db/seeds.rb:129]: Date can be changed to created_at, but still need to do Date.today - num

File: app/graphql/types/habit_type.rb [hunk 4]
```diff
@@ -5,5 +5,15 @@ class HabitType < Types::BaseObject
...
+    field :weekly_habit_entries, [Types::HabitEntryType]
+
+    def weekly_habit_entries
+      #if we add multiple users this needs to be changed, only adding it to maybe make the future easier
+      user = User.first
+      object.habit_entries
+            .where('user_id = ?', user.id)
+            .order(created_at: :desc)
+            .limit(7)
+    end
```
Comment (TannerDale) [inline app/graphql/types/habit_type.rb]: Is that where clause needed?

Comment (TannerDale) [reply app/graphql/types/habit_type.rb]: `user = User.find(user_id)`
`user.habit_entries.order(created_at: :desc).limit(7)`

**Review by Lmcourt — CHANGES_REQUESTED** [review]

File: spec/factories/habit_entries.rb [hunk 5]
```diff
@@ -1,8 +1,7 @@
 FactoryBot.define do
   factory :habit_entry do
-    date { Date.today }
-    status { [0, 1].sample }
-    association :user
-    association :habit
+    status { 1 }
+    user { nil }
+    habit { nil }
```
Comment (TannerDale) [inline spec/factories/habit_entries.rb]: These need to be reverted so we can use Faker

File: spec/requests/graphql/queries/fetch_user_weekly_habits_spec.rb (hunk 1 above)
Comment (TannerDale) [reply spec/requests/graphql/queries/fetch_user_weekly_habits_spec.rb]: It's just excited

**Comment by TannerDale** [comment]
Habit Entry factory was trying to create a date attribute that was removed and erroring out tests. I pulled down to differently  named branch and had to open a different request. Closing this PR but all changes are present in the other PR #26

**Pull request closed without merging**